alibi_detect/utils/random.py

```diff
@@ -0,0 +1,65 @@
+from contextlib import contextmanager
+import random
+import numpy as np
+import os
+from alibi_detect.utils.frameworks import has_pytorch, has_tensorflow
+
+if has_tensorflow:
+    import tensorflow as tf
+if has_pytorch:
+    import torch
+
+
+def set_seed(seed):
+    """
+    Sets the Python, NumPy, TensorFlow and PyTorch random seeds (if installed).
+    """
+    os.environ['PYTHONHASHSEED'] = str(seed)
+    random.seed(seed)
+    np.random.seed(seed)
+    if has_tensorflow:
+        tf.random.set_seed(seed)
+    if has_pytorch:
+        torch.manual_seed(seed)
+
+
+def get_seed() -> int:
+    """
+    Gets the seed set by :func:`set_seed`.
+
+    Example
+    -------
+    from alibi_detect.utils import set_seed, get_seed
+    set_seed(42)
+    get_seed()
+    >>> 42
+    """
+    return np.random.get_state()[1][0]  # type: ignore[index]
```

Comment:
Wonder if good enough to rely on `numpy` only for this but not the other libraries? Possible to get the state/seed from others too?

Reply:
Probably overkill/unnecessary but one could also have a global `alibi` variable which holds the seed set by `set_seed`.

Reply:
Just had a look and I could get the `tensorflow` and `torch` seeds with [tf.compat.v1.get_seed](https://www.tensorflow.org/api_docs/python/tf/compat/v1/get_seed) and [get_rng_state](https://pytorch.org/docs/stable/generated/torch.get_rng_state.html#torch.get_rng_state). What do you think we should do in the event of conflict though, just raise a warning saying that `set_seed` must have been called before `get_seed` can be used?